packages/next/src/app-dir/server.ts

```diff
@@ -25,35 +26,42 @@ import {
 import { revalidateTag } from 'next/cache';
 import { isNotFoundError } from 'next/dist/client/components/not-found';
 import { isRedirectError } from 'next/dist/client/components/redirect';
-import { cache } from 'react';
 import { formDataToObject } from './formDataToObject';
 import type {
   ActionHandlerDef,
-  CreateTRPCNextAppRouterOptions,
+  CreateTRPCNextAppRouterServerOptions,
   inferActionDef,
 } from './shared';
 import { generateCacheTag, isFormData } from './shared';
-import type { NextAppDirDecorateRouterRecord } from './types';
+import type {
+  NextAppDirDecorateRouterRecord,
+  NextAppDirRuntime,
+} from './types';
 
 // ts-prune-ignore-next
 export function experimental_createTRPCNextAppDirServer<
   TRouter extends AnyRouter,
->(opts: CreateTRPCNextAppRouterOptions<TRouter>) {
-  const getClient = cache(() => {
+>(opts: CreateTRPCNextAppRouterServerOptions<TRouter>) {
+  return createRecursiveProxy(async (callOpts) => {
     const config = opts.config();
-    return createTRPCUntypedClient(config);
-  });
+    const client = createTRPCUntypedClient(config);
```

Comment:
this can still be deduped as it was before, no?

Reply:
done!